mcs/tools/al/Al.cs

```diff
@@ -765,6 +729,83 @@ private void SetKeyPair (AssemblyName aname)
 			}
 		}
 
+		private void ReadCustomAttributesFromTemplateFile (string templateFile, ref AssemblyName aname)
```

Comment:
The `AssemblyName` instance gets populated before the call to this function, hence the use of `ref`.